src/Bundle/DependencyInjection/Compiler/PagerfantaBridgePass.php

```diff
@@ -48,12 +49,23 @@ private function aliasRenamedServices(ContainerBuilder $container): void
     {
         if ($container->hasDefinition('pagerfanta.twig_extension')) {
             $container->setAlias('twig.extension.pagerfanta', 'pagerfanta.twig_extension')
-                ->setDeprecated(true, 'The "%alias_id%" service alias is deprecated since Sylius 1.8, use the "pagerfanta.twig_extension" service ID instead.');
+                ->setDeprecated(...$this->getDeprecationMessage());
         }
 
         if ($container->hasDefinition('pagerfanta.view_factory')) {
             $container->setAlias('white_october_pagerfanta.view_factory', 'pagerfanta.view_factory')
-                ->setDeprecated(true, 'The "%alias_id%" service alias is deprecated since Sylius 1.8, use the "pagerfanta.view_factory" service ID instead.');
+                ->setDeprecated(...$this->getDeprecationMessage());
         }
     }
+
+    private function getDeprecationMessage(): array
+    {
+        $message = 'The "%alias_id%" service alias is deprecated since Sylius 1.8, use the "pagerfanta.view_factory" service ID instead.';
+
+        if (method_exists(Alias::class, 'getDeprecation')) {
```

Comment:
Won't it be better to base it on Symfony version?

Reply:
The `getDeprecation` method is added in the same PR, see https://github.com/symfony/symfony/pull/35778. So i think this should be good enough.

Reply:
I get it, I have the problem, that checking the number of arguments or Symfony version is a more explicit check for me. But perhaps, I'm just picky